HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiGuid.java

```diff
@@ -0,0 +1,415 @@
+ case "3b8c8162-188c-46a4-aec9-be43f1d65697":reference = "ux_capsule"; break;
+ case "605dab50-e046-4300-abb6-3dd810dd8b23":reference = "RH_Shim"; break;
+ case "8be4df61-93ca-11d2-aa0d-00e098032b8c":reference = "EFI_Global_Variable"; break;
+ case "91376aff-cba6-42be-949d-06fde81128e8":reference = "GRUB"; break;
+
+ // Partition Table GUIDs
+ case "0fc63daf-8483-4772-8e79-3d69d8477de4":reference = "Linux filesystem data"; break;
+ case "e6d6d379-f507-44c2-a23c-238f2a3df928":
+                reference = "Logical Volume Manager (LVM) partition"; break;
+ case "4f68bce3-e8cd-4db1-96e7-fbcaf984b709":reference = "Root partition (x86-64)"; break;
+ case "a19d880f-05fc-4d3b-a006-743f0f84911e":reference = "RAID partition "; break;
+ case "933ac7e1-2eb4-4f13-b844-0e14e2aef915":reference = "/home partition[ (x86-64)"; break;
+ case "ebd0a0a2-b9e5-4433-87c0-68b6b72699c7":reference = "GPT Basic data partition"; break;
+
+ // RHBoot Lenovo specific UUIDS
+ case "3cc24e96-22c7-41d8-8863-8e39dcdcc2cf":reference = "lenovo"; break;
+ case "82988420-7467-4490-9059-feb448dd1963":reference = "lenovo_me_config"; break;
+ case "f7e615b-0d45-4f80-88dc-26b234958560":reference = "lenovo_diag"; break;
+ case "665d3f60-ad3e-4cad-8e26-db46eee9f1b5":reference = "lenovo_rescue"; break;
+ case "721c8b66-426c-4e86-8e99-3457c46ab0b9":reference = "lenovo_setup"; break;
+ case "f46ee6f4-4785-43a3-923d-7f786c3c8479":reference = "lenovo_startup_interrupt"; break;
+ case "126a762d-5758-4fca-8531-201a7f57f850":reference = "lenovo_boot_menu"; break;
+ case "a7d8d9a6-6ab0-4aeb-ad9d-163e59a7a380":reference = "lenovo_diag_splash"; break;
+ // Company UUIDs (From Internet searches)
+ case "77fa9abd-0359-4d32-bd60-28f4e78f784b":reference = "Microsoft Inc."; break;
+ case "f5a96b31-dba0-4faa-a42a-7a0c9832768e":reference = "HPE Inc."; break;
+ case "2879c886-57ee-45cc-b126-f92f24f906b9":reference = "SUSE Certificate"; break;
+ case "70564dce-9afc-4ee3-85fc-949649d7e45c":reference = "Dell Inc."; break;
+
+ // Intel GUIDS
+ case "bfcc0833-2125-42d1-8c6d-13821e23c078":reference = "Intel(R) Desktop Boards"; break;
+ case "80b3ad5b-9880-4af9-a645-e56a68be89de":reference = "Intel(R) CISD FW Update"; break;
+
+// Microsoft GUIDS
+ case "e3c9e316-0b5c-4db8-817d-f92df00215ae":
+                reference = "Microsoft Reserved Partition (MSR)"; break;
+ case "5808c8aa-7e8f-42e0-85d2-e1e90434cfb3":
+                reference = "Logical Disk Manager (LDM) metadata partition "; break;
+ case "af9b60a0-1431-4f62-bc68-3311714a69ad":
+                reference = "Logical Disk Manager data partition"; break;
+ case "de94bba4-06d1-4d40-a16a-bfd50179d6ac":reference = "Windows Recovery Environment"; break;
+ case "9f25ee7a-e7b7-11db-94b5-f7e662935912":reference = "Windows Boot Loader"; break;
+
+ // Linux specific GUIDS
+ case "0fc63daf-8483-4772-8e79-3d69d8477de":reference = "Linux filesystem data"; break;
+ case "44479540-f297-41b2-9af7-d131d5f0458a4":reference = "Root partition (x86)"; break;
```

Comment:
This has a different length. Is this the correct full ID?

Reply:
extra 4 at the end in this one.